openedx/core/djangoapps/edx_global_analytics/tasks.py

```diff
@@ -3,70 +3,79 @@
 and send this data to appropriate service for further processing.
 """
 
-import datetime
 import json
+import logging
 
 import requests
 from celery.task import task
 
 from django.conf import settings
 from django.contrib.sites.models import Site
-from django.db.models import Count
-from django.db.models import Q
 
 from xmodule.modulestore.django import modulestore
-from student.models import UserProfile
 from .models import TokenStorage
 
+from .utils import (
+    fetch_instance_information,
+    get_previous_day_start_and_end_dates,
+    get_previous_week_start_and_end_dates,
+    get_previous_month_start_and_end_dates,
+    cache_timeout_week,
+    cache_timeout_month
+)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+def paranoid_level_statistics_bunch():
+    active_students_amount_day = fetch_instance_information(
+        'active_students_amount_day', 'active_students_amount',
+        get_previous_day_start_and_end_dates(), cache_timeout=None
+    )
+
+    active_students_amount_week = fetch_instance_information(
+        'active_students_amount_week', 'active_students_amount',
+        get_previous_week_start_and_end_dates(), cache_timeout_week()
+    )
+
+    active_students_amount_month = fetch_instance_information(
+        'active_students_amount_month', 'active_students_amount',
+        get_previous_month_start_and_end_dates(), cache_timeout_month()
+    )
+
+    return active_students_amount_day, active_students_amount_week, active_students_amount_month
+
+
+def enthusiast_level_statistics_bunch():
+    students_per_country = fetch_instance_information(
+        'students_per_country', 'students_per_country',
+        get_previous_day_start_and_end_dates(), cache_timeout=None
+    )
+
+    return students_per_country
+
 
 @task
 def count_data():
     """
-    Periodic task function that gathers information about the students amount,
-    geographical coordinates of the platform, courses amount and
-    makes a POST request with the data to the appropriate service.
+    Periodic task function that gathers instance information as like as platform name,
+    active students amount, courses amount etc., then makes a POST request with the data to the appropriate service.
+
+    Sending information depends on statistics level in settings, that have an effect on bunch of data size.
     """
 
     # OLGA settings
     olga_settings = settings.ENV_TOKENS.get('OPENEDX_LEARNERS_GLOBAL_ANALYTICS')
```

Comment:
What if settings are not set in config file? Do you handle this situation or task will crash?